tools/src/verdaccio/registry.ts

```diff
@@ -89,92 +91,108 @@ export function nxStartVerdaccioServer({
+  }
+
+  return (
+    new Promise<RegistryResult>((resolve, reject) => {
+      executeProcess({
+        command: 'npm',
+        args,
+        shell: true,
+        observer: {
+          onStdout: (stdout: string) => {
+            if (verbose) {
+              process.stdout.write(
+                `${gray('>')} ${gray(bold('Verdaccio'))} ${stdout}`,
+              );
+            }
+
+            // Log of interest: warn --- http address - http://localhost:<PORT-NUMBER>/ - verdaccio/5.31.1
+            if (!startDetected && stdout.includes('http://localhost:')) {
+              // only setup env one time
+              startDetected = true;
+
+              const result: RegistryResult = {
+                registry: {
+                  storage,
+                  ...parseRegistryData(stdout),
+                },
+                // https://verdaccio.org/docs/cli/#default-database-file-location
+                stop: () => {
+                  // this makes the process throw
+                  killProcesses({ commandMatch: commandId });
+                },
+              };
+
+              console.info(
+                `${gray('>')} ${gray(
+                  bold('Verdaccio'),
+                )} Registry started on URL: ${bold(
+                  result.registry.url,
+                )}, with PID: ${bold(
+                  listProcess({ commandMatch: commandId }).at(0)?.pid,
+                )}`,
+              );
+              if (verbose) {
+                console.info(`${gray('>')} ${gray(bold('Verdaccio'))}`);
+                console.table(result);
+              }
+
+              resolve(result);
+            }
+          },
+          onStderr: (stderr: string) => {
+            if (verbose) {
+              process.stdout.write(
+                `${red('>')} ${red(bold('Verdaccio'))} ${stderr}`,
+              );
+            }
+          },
         },
-        onStderr: (data: string) => {
-          if (verbose) {
-            process.stdout.write(data);
+      })
+        // @TODO reconsider this error handling
+        .catch(error => {
+          if (error.message !== 'Failed to start verdaccio: undefined') {
+            console.error(
+              `${red('>')} ${red(
+                bold('Verdaccio'),
+              )} Error starting ${projectName} verdaccio registry:\n${error}`,
+            );
+          } else {
+            reject(error);
           }
-        },
-      },
+        });
     })
-      // @TOSO reconsider this error handling
-      .catch(error => {
-        if (error.message !== 'Failed to start verdaccio: undefined') {
-          console.error(
-            `Error starting ${projectName} verdaccio registry:\n${
-              error as Error
-            }`,
-          );
-          reject(error);
-        }
-        teardownTestFolder(storage);
-        throw error;
-      });
-  });
+      // in case the server dies unexpectedly clean folder
+      .catch(() => teardownTestFolder(storage))
```

Comment:
Careful not to swallow the error.

```suggestion
      .catch((error: unknown) => {
        teardownTestFolder(storage);
        throw error;
      })
```